sky/packages/sky/lib/src/rendering/box.dart

```diff
@@ -45,24 +67,14 @@ class BoxConstraints extends Constraints {
       minHeight = height != null ? height : 0.0,
       maxHeight = height != null ? height : double.INFINITY;
 
+  /// Constructs a box constraints that forbids sizes larger than the given size
   BoxConstraints.loose(Size size)
     : minWidth = 0.0,
       maxWidth = size.width,
       minHeight = 0.0,
       maxHeight = size.height;
 
-  const BoxConstraints.expandWidth({
-    this.maxHeight: double.INFINITY
-  }): minWidth = double.INFINITY,
-      maxWidth = double.INFINITY,
-      minHeight = 0.0;
-
-  const BoxConstraints.expandHeight({
-    this.maxWidth: double.INFINITY
-  }): minWidth = 0.0,
-      minHeight = double.INFINITY,
-      maxHeight = double.INFINITY;
-
+  /// A box constraints that requires an infinite width and height
```

Comment:
rather than defining this in terms of infinite width and height, explain it in terms of being expected to be constrained by another BoxConstraints, so as to select the maximum widths and heights of those constraints.